maven-release-manager/src/main/java/org/apache/maven/shared/release/transform/jdom2/JDomModel.java

```diff
@@ -198,7 +199,17 @@ public void setVersion(String version) {
                         AbstractRewritePomsPhase.extractPropertyFromExpression(versionElement.getTextNormalize());
                 Properties properties = getProperties();
                 if (properties != null) {
-                    properties.computeIfPresent(ciFriendlyPropertyName, (k, v) -> version);
+                    String sha1 = properties.getProperty("sha1", "");
+                    String changelist = properties.getProperty("changelist", "");
+                    properties.setProperty(
+                            ciFriendlyPropertyName,
+                            // assume that everybody follows the example and properties are simply chained
+                            version.replaceAll(sha1, "").replaceAll(changelist, ""));
```

Comment:
This is a regex, where is the guarantee not requiring escaping?

Reply:
I still consider using a regex method wrong here...

Reply:
Addressed